### PR TITLE
Handle shift and CAPS keys

### PR DESCRIPTION
**Pull request by fjaguado**
# PR Details

This PR enables the shift key only for the next pressed key and returns to the default layout after that.

## Related Issue

#3 

## How Has This Been Tested

The included tests have been executed

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation 
- [x] I have updated the documentation accordingly (README.md for each UI component)
- [x] I have added tests to cover my changes (at least 1 spec for each UI component with the same coverage as the master branch)
- [x] All new and existing tests passed
- [x] A new branch needs to be created from master to evolve previous versions
- [x] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] All UI components must be added into the showcase (at least 1 component with the default settings)
- [x] Add the issue into the right [project](https://github.com/systelab/systelab-components/projects) with the proper status (In progress)

### CODE REVIEW

File: projects/showcase/test/specs/alphanumeric-layout.spec.ts [hunk 1]
```diff
@@ -69,15 +69,7 @@ describe("AlphanumericLayout", () => {
...
-        await ReportUtility.addExpectedResult("Virtual keyboard now shows all letters in uppercase and more symbols", async() => {
-            expect(await VirtualKeyboard.get().isPresent()).toBeTruthy();
-            expect(await VirtualKeyboard.get().getRowCount()).toEqual(5);
-            expect(await VirtualKeyboard.get().getRowKeys(0)).toEqual([      '~', '!', '@', '#', '$', '%', '^', '&amp;', '*', '(', ')', '_', '+', '{bksp}']);
-            expect(await VirtualKeyboard.get().getRowKeys(1)).toEqual([  '{tab}', 'Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P', '{', '}', '|']);
-            expect(await VirtualKeyboard.get().getRowKeys(2)).toEqual([ '{lock}', 'A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L', ':', '"', '{enter}']);
-            expect(await VirtualKeyboard.get().getRowKeys(3)).toEqual(['{shift}', 'Z', 'X', 'C', 'V', 'B', 'N', 'M', '&lt;', '&gt;', '?', '{shift}']);
-            expect(await VirtualKeyboard.get().getRowKeys(4)).toEqual(['{space}']);
-        });
+        await expectAlphanumericUppercaseLayout();
```
Comment (joaquimvila) [inline projects/showcase/test/specs/alphanumeric-layout.spec.ts:72]: I am OK on moving these expectations into a shared method to faciliate reusing it. 
However, I would suggest to create a dedicated class for that.
I would suggest to place them into a `expectation/layout.expectation.ts` file in a class named `LayoutExpectation`. Then, the methods should be converted to `static`, so we don't need to create a class for them.

Comment (joaquimvila) [reply projects/showcase/test/specs/alphanumeric-layout.spec.ts:72]: Also for consistency, the numeric layout expectation should be placed there too.

File: projects/showcase/test/specs/alphanumeric-layout.spec.ts [hunk 2]
```diff
@@ -101,15 +93,17 @@ describe("AlphanumericLayout", () => {
...
-        });
+        await expectAlphanumericLowercaseLayout();
+    });
+
+    it("Click on 'Shift' virtual keyboard key to change layout to uppercase for only once key", async () => {
+        await VirtualKeyboard.get().clickShift();
+
+        await expectAlphanumericUppercaseLayout();
+
+        await VirtualKeyboard.get().clickKeys('S');
```
Comment (joaquimvila) [inline projects/showcase/test/specs/alphanumeric-layout.spec.ts:104]: As we are clicking a SINGLE key, we could use the `clickKey` method

File: projects/showcase/test/specs/alphanumeric-layout.spec.ts [hunk 3]
```diff
@@ -101,15 +93,17 @@ describe("AlphanumericLayout", () => {
...
-            expect(await VirtualKeyboard.get().getRowKeys(4)).toEqual(['{space}']);
-        });
+        await expectAlphanumericLowercaseLayout();
+    });
+
+    it("Click on 'Shift' virtual keyboard key to change layout to uppercase for only once key", async () => {
+        await VirtualKeyboard.get().clickShift();
+
+        await expectAlphanumericUppercaseLayout();
+
```
Comment (joaquimvila) [inline projects/showcase/test/specs/alphanumeric-layout.spec.ts:103]: The code is OK, but to keep the actions vs expected results separation, the click on the S key and the later expectation should be placed on a separated "it".

File: projects/systelab-virtual-keyboard/src/lib/systelab-virtual-keyboard.component.ts [hunk 4]
```diff
@@ -221,6 +223,11 @@ export class SystelabVirtualKeyboardComponent implements AfterViewInit {
     }
 
     this.dispatchEvents(button);
+
+    if (this.shiftPressed) {
+      this.toggleShift();
```
Comment (joaquimvila) [inline projects/systelab-virtual-keyboard/src/lib/systelab-virtual-keyboard.component.ts:228]: The name of this method would be better as `toggleLayout()` as what we are toggling is the layout, not the shift.

**Review by joaquimvila — CHANGES_REQUESTED** [review]
Good job @fjaguado with this new feature. Requested some changes, specially on the test automation, but looks good in overall.

File: projects/systelab-virtual-keyboard/src/lib/systelab-virtual-keyboard.component.ts (hunk 4 above)
Comment (fjaguado) [reply projects/systelab-virtual-keyboard/src/lib/systelab-virtual-keyboard.component.ts:228]: No, we're are toggling Shift. I mean, we're simulating a shift key pulse

File: projects/showcase/test/specs/alphanumeric-layout.spec.ts (hunk 2 above)
Comment (fjaguado) [reply projects/showcase/test/specs/alphanumeric-layout.spec.ts:104]: I didn't see the clickKey method. Changed!

File: projects/systelab-virtual-keyboard/src/lib/systelab-virtual-keyboard.component.ts (hunk 4 above)
Comment (fjaguado) [reply projects/systelab-virtual-keyboard/src/lib/systelab-virtual-keyboard.component.ts:228]: I changed to toggleShifLayout

**Pull request merged**